A-New-BellHope/bellhopcuda
# Dll build

Generates a dll build for the ray case of the non-cuda version (only) along with the cuda and command line versions. 
Developed with VS 2022 and cuda 11.6 (make work with other combinations, but CMake support is much improved in 2022 and it needs cuda 11.6 for support).

To help with testing I also made a simple executable to load/run the dll [here](https://github.com/A-New-BellHope/bellhopdll_test)

To test:

1. Download and build the release for this, check bin folder for bellhopcxx.exe, bellhopcuda.exe, and bellhopcxxdll.dll and bellhopcxxdll.lib  (last two are new).
2. Check that the cli versions work as before (run on a random file, or even do the full test suite if you're feeling zesty).
3. either roll your own dll import/test or copy bellhopcxxdll.dll/lib to testbellhopdll/testbellhopdll (next to main.cpp) and compile/run that. Bear in mind, it only works with rays for now.
